packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,6 +293,39 @@ export class CfnInclude extends core.CfnElement {
     this.conditions[conditionName] = cfnCondition;
   }
 
+  private createNestedStack(nestedStackId: string, props?: CfnIncludeProps): void {
+    if (nestedStackId in this.template.Resources) {
+      if (this.template.Resources[nestedStackId].Type === 'AWS::CloudFormation::Stack' && props?.nestedStacks) {
+        const nestedStackScope = new core.NestedStack(this, 'Nested' + nestedStackId, {
```

Comment:
Why are you adding `'Nested'` to the ID of the Stack? That means if you have `ChildStack` nested stack and `NestedChildStack` resource, it will conflict, no?

Those should be created with just their `nestedStackId`.